argo/providers/vai/provider.go

```diff
@@ -388,17 +464,25 @@ func (vaip VAIProvider) EnqueueRun(ctx context.Context, providerConfig VAIProvid
 	return vaiRun.RunId, nil
 }
 
+func extractBucketAndObjectFromGCSPath(gcsPath string) (string, string, error) {
+	r := regexp.MustCompile(`gs://([^/]+)/(.+)`)
+	matched := r.FindStringSubmatch(gcsPath)
+	if len(matched) < 3 {
+		return "", "", errors.New(fmt.Sprintf("invalid gs URI %s", gcsPath))
+	}
+	return matched[1], matched[2], nil
+}
+
 func (vaip VAIProvider) specFromTemplateUri(ctx context.Context, providerConfig VAIProviderConfig, job *aiplatformpb.PipelineJob) error {
```

Comment:
I think we should rename this to make it clear that this mutates the object passed into it

Reply:
Suggested name 🤔 `mutateJobWith....?`

Reply:
Have gone with "enrich"